tree/20_cfe_basics/cfengine/packages.cf

```diff
@@ -99,6 +99,19 @@ bundle common rpm_knowledge
       "rpm3_arch_regex" string => "\S+\s+(\S+).*";
 }
 
+bundle common redhat_no_locking_knowledge {
+  vars:
+    !redhat_setopt_not_available::
```

Comment:
This class is sensitive to normal ordering. In my tests, the result of the real test below didn't matter, this case was always set. I strongly recommend adding a "check_done" class below and prefixing it to this condition.